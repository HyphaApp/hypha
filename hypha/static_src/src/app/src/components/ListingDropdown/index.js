import React from 'react';
import PropTypes from 'prop-types';
import smoothscroll from 'smoothscroll-polyfill';

export default class ListingDropdown extends React.Component {
    static propTypes = {
        listRef: PropTypes.object,
        groups: PropTypes.array,
        scrollOffset: PropTypes.number,
    }

    componentDidMount() {
        // polyfill element.scrollTo
        smoothscroll.polyfill();
    }

    handleChange(e) {
        const groupHeaderPosition = document.getElementById(e.target.value).offsetTop - this.props.scrollOffset;

        this.props.listRef.current.scrollTo({
            top: groupHeaderPosition
        })
    }

    renderListDropdown() {
        const { groups } = this.props;
        return (
            <form className="form form__select">
                <select onChange={(e) => this.handleChange(e)} aria-label="Jump to listing group">
                    {groups.map(group => {
                        return (
                            <option key={`listing-heading-${group.key}`} value={group.key}>{group.name}</option>
                        )
                    })}
                </select>
            </form>
        )
    }

    render() {
        return (
            <>
                {this.renderListDropdown()}
            </>
        )
    }
}
