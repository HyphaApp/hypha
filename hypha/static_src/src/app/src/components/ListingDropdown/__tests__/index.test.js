import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import ListingDropdown from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test listing drop down component", () => {

    const listRef = { current : 'ul.listing.listing--applications'} 
    const groups = [{
                  items: [1,2],
                  key: "round-37",
                  name: "review api test"
                }]
    const scrollOffset = 75

    const subject = mount(<ListingDropdown
        listRef={listRef}
        groups={groups}
        scrollOffset={scrollOffset}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.form__select').length).toBe(1);
      expect(subject.find('option').length).toBe(1)
      expect(subject.find('select').children().length).toBe(1)
      expect(subject.find('option').text()).toEqual(groups[0].name)
    });

    // it("test methods ", () => {
    //   const spy = jest.spyOn(ListingDropdown.prototype, 'handleChange');
    //   const wrapper = mount(<ListingDropdown listRef={listRef}
    //     groups={groups}
    //     scrollOffset={scrollOffset} />);
    //   wrapper.instance().handleChange({target: {value : 1}});
    //   expect(spy).toHaveBeenCalled();
    // });
  
    test('render a listing drop down component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
