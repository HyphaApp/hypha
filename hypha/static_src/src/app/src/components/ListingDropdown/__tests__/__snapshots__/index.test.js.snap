// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test listing drop down component render a listing drop down component 1`] = `
<ListingDropdown
  groups={
    Array [
      Object {
        "items": Array [
          1,
          2,
        ],
        "key": "round-37",
        "name": "review api test",
      },
    ]
  }
  listRef={
    Object {
      "current": "ul.listing.listing--applications",
    }
  }
  scrollOffset={75}
>
  <form
    className="form form__select"
  >
    <select
      aria-label="Jump to listing group"
      onChange={[Function]}
    >
      <option
        key="listing-heading-round-37"
        value="round-37"
      >
        review api test
      </option>
    </select>
  </form>
</ListingDropdown>
`;
